### ISSUE TITLE
[version 3.8] hide() immediately removes custom class names when transition starts, causes jarring styling reset before transition ends. (possible solution attached)

### ISSUE THREAD

**Issue by cheapsteak** [fabien-d/alertify.js#130]
hide() immediately removes custom class names when transition starts, causes jarring styling reset before transition ends. (possible solution attached)

Issue: 
Dialog boxes created with `alertify.confirm(msg, fn, cssClassName)` gets the custom class name stripped when the hide() function replaces the dialog's className attribute, makes the dialog to reset its appearance before it's actually hidden.

Possible solution:
Have hide() append `alertify-hide alertify-hidden` to the className attribute instead of replacing it, and reset the className attribute on `transitionDone`

``` javascript
transitionDone = function (event) {
  event.stopPropagation();
  //elDialog.className += " alertify-isHidden"; //removed this line
  elDialog.className = "alertify alertify-hidden alertify-isHidden"; //added this line
  self.unbind(elDialog, self.transition, transitionDone);
};
if (typeof this.transition !== "undefined") {
  this.bind(elDialog, this.transition, transitionDone);
  //elDialog.className = "alertify alertify-hide alertify-hidden"; //removed this line
  elDialog.className += " alertify-hide alertify-hidden"; //added this line
} else {
  //...
```

One potential problem that this causes is that the dialog box would have both `alertify-show` and `alertify-hide` classes at the same time.  
It doesn't seem to cause weird behavior, but I haven't done much testing. There's probably a more elegant solution, but this works for now.

Really love this plugin btw.

**Comment by fabien-d** [comment]
Good catch... agreed that I don't really like having `-show` and `-hide` at the same time. Will look for a more elegant solution.
